tests/integration/deploying/deploying_contract_test.py

```diff
@@ -0,0 +1,57 @@
+# pylint: disable=consider-using-with
+import subprocess
+import time
+from distutils.file_util import copy_file
+from pathlib import Path
+from socket import socket as Socket
+
+import pytest
+
+from protostar.commands.deploy.deploy_contract import deploy_contract
+
+
+def get_available_port() -> int:
+    with Socket() as socket:
+        socket.bind(("", 0))
+        return socket.getsockname()[1]
+
+
+@pytest.fixture(name="devnet_gateway_url")
+def devnet_gateway_url_fixture():
```

Comment:
I like the approach, but I think we should set scope to `session` as startup is quite expensive